lib/bundler/shared_helpers.rb

```diff
@@ -207,6 +222,14 @@ def clean_load_path
       end
     end
 
+    def prints_major_deprecations?
+      require "bundler"
+      return false unless Bundler.settings[:major_deprecations]
```

Comment:
I would strongly prefer that this be a constant, maybe even in `bundler/version.rb` so that we can only require the file with the constant. This isn't a setting we want to change often.